hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconContainerManager.java

```diff
@@ -122,26 +122,87 @@ public void checkAndAddNewContainer(ContainerID containerID,
           scmClient.getContainerWithPipeline(containerID.getId());
       LOG.debug("Verified new container from SCM {}, {} ",
           containerID, containerWithPipeline.getPipeline().getId());
-      // If no other client added this, go ahead and add this container.
-      if (!containerExist(containerID)) {
-        addNewContainer(containerID.getId(), containerWithPipeline);
-      }
+      // no need call "containerExist" to check, because
+      // 1 containerExist and addNewContainer can not be atomic
+      // 2 addNewContainer will double check the existence
```

Comment:
If we finally adopt the batch solution, can we drop this singular method ?

Reply:
thanks for the review!  maybe we can  maintain it here for the present, and remove it later in another patch

Reply:
I think the original check is kind of optimistic locking here?

Reply:
@symious Thank for the review!  "addNewContainer" will call "ContainerStateMap#contains" ,and this is what "containerExist" exactly does, so no need here.

Reply:
There are still some steps that need to be taken before `addNewContainer` calls `ContainerStateMap$contains`, with the contains check here, the performance would be better.
Besides, in `ContainerStateManagerImpl`, the writeLock would be required to check if the container exists, it's not reasonable to involve the lock here.

Reply:
"check the existence and add" isn't meant to be an atomic operation here, it's just used to avoid some unnecessary operations if the other clients have updated the containerId already.